pkg/etcd/etcd.go

```diff
@@ -1048,18 +1051,62 @@ func (e *ETCD) manageLearners(ctx context.Context) {
 			continue
 		}
 
+		nodes, err := e.parseNodesWithEtcd()
+		if err != nil {
+			logrus.Errorf("Error while listing nodes with etcd status: %v", err)
+			continue
+		}
+
 		for _, member := range members.Members {
 			if member.IsLearner {
 				if err := e.trackLearnerProgress(ctx, progress, member); err != nil {
 					logrus.Errorf("Failed to track learner progress towards promotion: %v", err)
 				}
+
+				// If the node exists in the parse and is a learner, then the status condition is set.
+				if node, ok := nodes[member.Name]; ok {
+					_, err := e.setEtcdStatusCondition(&node, false)
+					if err != nil {
+						logrus.Errorf("Unable to set etcd status condition %s: %v", member.Name, err)
+					}
+				}
+
 				break
 			}
+
+			// If the node exists in the parse and is a voter, then the status condition is set.
+			if node, ok := nodes[member.Name]; ok {
+				_, err := e.setEtcdStatusCondition(&node, true)
+				if err != nil {
+					logrus.Errorf("Unable to set etcd status condition %s: %v", member.Name, err)
+				}
+			}
 		}
 	}
 	return
 }
 
+// parseNodesWithEtcd searches for nodes that have an annotation with etcd.{version.Program}.cattle.io/node-name
+// and sets a hashMap for easy node retrieval by node name.
+func (e *ETCD) parseNodesWithEtcd() (map[string]v1.Node, error) {
+	newMap := map[string]v1.Node{}
+
+	nodes := e.config.Runtime.Core.Core().V1().Node()
+	etcdSelector := labels.Set{util.ETCDRoleLabelKey: "true"}
+	nodeList, err := nodes.List(metav1.ListOptions{LabelSelector: etcdSelector.String()})
+	if err != nil {
+		return newMap, err
+	}
+
+	for _, node := range nodeList.Items {
+		if etcdName, ok := node.Annotations["etcd."+version.Program+".cattle.io/node-name"]; ok {
```

Comment:
You don't need this any more. The labelSelector will ensure that the list only returns etcd nodes; you should not additionally filter the node list by the annotation.

Reply:
I've was using annotations because I saw member list and the node have different names when you change the node name after having set up etcd previously. 

if you follow this steps here

```bash
k3s server --cluster-init
```

wait a little bit, kill the process and 

```bash
k3s server --cluster-init --node-name tests
```

Reply:
I believe changing the node name like this is not a good practice, so how we can avoid this?